metamask-predicate/tests/harness.rs

```diff
@@ -0,0 +1,204 @@
+use fuel_vm::fuel_asm::{op, RegId};
+use fuels::{
+    prelude::*,
+    test_helpers::Config,
+    tx::{Address, AssetId, Contract, Output},
+    types::{
+        input::Input,
+        resource::Resource,
+        transaction_builders::{ScriptTransactionBuilder, TransactionBuilder},
+        unresolved_bytes::UnresolvedBytes,
+        Token,
+        Bits256,
+        B512,
+    },
+};
+use std::str::FromStr;
+
+async fn setup() -> (Vec<u8>, Address, WalletUnlocked, u64, AssetId) {
+    let predicate_code =
+        std::fs::read("./out/debug/metamask-predicate.bin")
+            .unwrap();
+    let predicate_address = (*Contract::root_from_code(&predicate_code)).into();
```

Comment:
I believe the rust sdk provides an interface for loading predicates such that we do not need to use `fs` ourselves (directly).